tools/QuietBackgroundProcesses/DevHome.QuietBackgroundProcesses.PerformanceRecorderEngine/PerformanceRecorderEngine.cpp

```diff
@@ -0,0 +1,669 @@
+    L"OfficeClickToRun.exe",
+    L"Microsoft.SharePoint.exe",
+    L"msedge.exe",
+    L"msedgewebview2.exe",
+    L"ShellExperienceHost.exe",
+    L"StartMenuExperienceHost.exe",
+    L"smartscreen.exe",
+    L"sihost.exe",
+    L"SystemSettings.exe",
+    L"electron.exe",
+    L"CrmSandbox.exe",
+    L"ms-teams.exe",
+    L"TextInputHost.exe",
+    L"UserOOBEBroker.exe",
+    L"WebViewHost.exe",
+    L"Widgets.exe",
+    L"WidgetService.exe",
+    L"XboxGameBarWidgets.exe",
+    L"teams.exe"
+};
+std::set<std::wstring> c_system = {
+    L"System",
+    L"Registry",
+    L"Secure System",
+    L"audiodg.exe",
+    L"ctfmon.exe",
+    L"LogonUI.exe",
+    L"MpDefenderCoreService.exe",
+    L"MpDlpService.exe",
+    L"ShellHost.exe",
+    L"smss.exe",
+    L"spoolsv.exe",
+    L"wininit.exe",
+    L"lsass.exe"
+};
+std::set<std::wstring> c_developer = {
+    L"cmd.exe",
+    L"conhost.exe",
+    L"console.exe",
+    L"OpenConsole.exe",
+    L"powershell.exe",
+    L"cl.exe",
+    L"link.exe",
+    L"devenv.exe",
+    L"DevHome.exe",
+    L"DevHomeGitHubExtension.exe",
+    L"python.exe",
+    L"build.exe",
+    L"msbuild.exe",
+    L"windbg.exe",
+    L"windbgx.exe",
+    L"EngHost.exe",
+    L"DbgX.Shell.exe",
+    L"GVFS.Mount.exe",
+    L"GVFS.Service.exe",
+    L"GVFS.ServiceUI.exe",
+    L"vscode.exe",
+    L"code.exe",
+    L"cpptools.exe",
+    L"notepad.exe",
+    L"notepad++.exe",
+    L"Wex.Services.exe",
+    L"Taskmgr.exe",
+    L"wpa.exe",
+    L"wpr.exe",
+    L"CalculatorApp.exe",
+    L"npm.exe",
+    L"winget.exe",
+    L"chocolatey.exe",
+    L"pip.exe",
+    L"vshost.exe",
+    L"VSSVC.exe",
+    L"VBCSCompiler.exe",
+    L"vcpkgsrv.exe",
+    L"WindowsTerminal.exe",
+    L"WindowsPackageManagerServer.exe",
+    L"reSearch.exe"
+};
+std::set<std::wstring> c_vms = {
+    L"vmmem",
+    L"vmwp.exe",  // actual process name for 'vmmem'
+    L"vmcompute.exe",
+    L"vmconnect.exe",
+    L"vmwp.exe",
+    L"vmms.exe"
+};
+std::set<std::wstring> c_background = {
+    L"services.exe",
+    L"svchost.exe",
+    L"SCNotification.exe",
+    L"SecurityHealthyService.exe",
+    L"DevHome.QuietBackgroundProcesses.Server.exe",
+    L"DevHome.QuietBackgroundProcesses.ElevatedServer.exe",
+    L"OneDrive.exe",
+    L"MsMpEng.exe",
+    L"MsSense.exe",
+    L"NdrSetup.exe",
+    L"NisSrv.exe",
+    L"RuntimeBroker.exe",
+    L"rundll32.exe",
+    L"SearchHost.exe",
+    L"SenseCE.exe",
+    L"SenseNdr.exe",
+    L"SenseNdrX.exe",
+    L"SenseTVM.exe",
+    L"SearchIndexer.exe",
+    L"taskhostw.exe",
+    L"winlogon.exe",
+};
+
+ProcessCategory GetCategory(DWORD pid, std::wstring_view processName)
+{
+    auto search = [&](std::wstring_view processName, const auto& list) {
+        auto it = std::find_if(list.begin(), list.end(), [&](const auto& elem) {
+            return wil::compare_string_ordinal(processName, elem, true) == 0;
+            });
+        auto found = (it != list.end());
+        return found;
+    };
+
+    if (pid == 4)
```

Comment:
Why is 4 significant?

Reply:
added comment
'PID 4 is the System process'